### ISSUE TITLE
PDAL: readers.text: Duplicate dimension 'X' detected in input file

### ISSUE THREAD

**Issue by nomeata** [PDAL/PDAL#3859]
**Describe the bug**

I would have expected that the following pipeline reads points from a file, then creates a translated clone of these points, and writes them all together to a new file:

Pipeline:
```
[
  { "type":"readers.text", "filename": "test.csv", "tag":"track"},
  { "type":"filters.transformation", "inputs": ["track"], "matrix": "1 0 0 1 0 1 0 0  0 0 1 0 0 0 0 1", "tag": "track_copy" },
  { "type":"filters.merge", "inputs": ["track", "track_copy"]},
  "test-out.txt"
]
```

Data:
```
"X","Y","Z"
1,1,1

```

Command:
```
pdal pipeline test-pipeline.json 
```

Output:
```
PDAL: readers.text: Duplicate dimension 'X' detected in input file 'test.csv'.
```

**Expected behavior**
I’d expect this to write a file with coordinates
```
1,1,1
2,1,1
```

Maybe it is simply wrong to use a stage twice as input, and every stage must be consumed exactly once? But in that case the error message should indicate that (and maybe https://pdal.io/en/2.4.3/pipeline.html#stage-objects could indicate such restrictions).

(Also I appreciate hints as to better ways to duplicate points – I need to make my point cloud more dense by replacing each point with a few points translated into various directions.)

**System/installation information:**
```
$ uname -a
Linux riviera 5.19.1 #1-NixOS SMP PREEMPT_DYNAMIC Thu Aug 11 11:22:05 UTC 2022 x86_64 GNU/Linux
```

```
$ pdal --version
-----------------------------------------------------------------------------------------------
pdal 2.4.0 (git-version: Release)
-----------------------------------------------------------------------------------------------
```

**Comment by abellgithub** [comment]
This is just a bug in the text reader and has nothing to do with your usage. I'll fix shortly.

**Comment by abellgithub** [comment]
In the meantime, a couple of things that might help:
- This is only an issue with text files where the dimension names are quoted. Remove the quotes and things should work.
- You can duplicate points by simply listing the input file multiple times. This should also get around the issue you're having.

**Comment by nomeata** [comment]
Thanks! I think I tried reading it twice, but got the same error. But not sure, I'll check when I'm at the PC again.

**Comment by hobu** [comment]
Fixed in #3860
